src/meta/processors/admin/Balancer.cpp

```diff
@@ -446,24 +465,27 @@ Balancer::buildLeaderBalancePlan(HostLeaderMap* hostLeaderMap, GraphSpaceID spac
     std::unordered_map<HostAddr, std::vector<PartitionID>> allHostParts;
     getHostParts(spaceId, allHostParts, totalParts);
 
-    size_t avgLoad = leaderParts / hostLeaderMap->size();
+    std::unordered_set<HostAddr> activeHosts;
```

Comment:
What's the purpose?

Reply:
Previously we balance leader between all active hosts, but when we add a new machine (it has no valid partition), we should not consider it.